app/code/Magento/Sales/Api/OrderHistoryInterface.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+// @codingStandardsIgnoreFile
+namespace Magento\Sales\Api;
+/**
+ * Order status history interface.
+ *
+ * An order is a document that a web store issues to a customer. Magento generates a sales order
+ * that lists the product items, billing and shipping addresses, and shipping and payment methods.
+ * A corresponding external document, known as a purchase order, is emailed to the customer.
+ *
+ * @api
+ * @since 101.0.2
+ */
+interface OrderHistoryInterface
```

Comment:
I see a couple of problems with this interface: 
1. Interface takes a lot of responsibility, it aggregates business logic for different entities: invoice, shipment, creditmemo, etc. Better solution - use separate interfaces for each entity (`CreditmemoHistoryRepositoryInterface`, `ShipmentHistoryRepositoryInterface`, etc.)
2. Better use objects in return types or array of objects.
3. Strict types required for new implementation. 
4. Unnecessary @since tag, it will be added automatically 
5. Better pass to method entityId and retrieve it from the repository
6. I don't understand the main purpose of this interface, it's repository, management or something else? 

Summarize described above I may conclude, that in current case we don't need the interface because it looks like syntax sugar for retrieving information from order entity, I think extract to separate object without interface will be enough.

Reply:
@sidolov thank you for your review! I agree with you on all points. The main idea of it was just to use it like interface for better composition purpose. In other words just to have an ability to operate with "rewriting" as it declares in Magento2 way :) But I see that in current case we really do not need it. I will remove it.